examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,258 @@
+
+model = ShallowWaterModel(
+    architecture=CPU(),
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions have a small perturbation that is random in space and 
+# decays away from the center of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2);
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 .+ ϵ * exp(- (y - Ly/2)^2 ) * randn();
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2);
+ hⁱ(x, y, z) = model.grid.Lz .+ ηⁱ(x, y, z);
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z);
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
+   ω_pert = ComputedField( ω_field - Ω )
+
+# Progress will output the clock times and the norm of the cross channel velocity,
+# as this is used to compute the growth rates.  We obtain the `norm` function from `LinearAlgebra`.
+
+using LinearAlgebra, Printf
+
+function progress(sim)
+    compute!(v)
+    @printf("Iteration: %d, time: %s, norm v: %f\n",
+    sim.model.clock.iteration,
+    prettytime(sim.model.clock.time),
+    norm(interior(v)) )
+end
+
+# ## Running a `Simulation`
+#
+# We pick the time-step that ensures to resolve the surface gravity waves.
+# A time-step wizard can be applied to use an adaptive time step.
+
+simulation = Simulation(model, Δt = 1e-2, stop_time = 150.0)
+
+# ## Prepare output files
+#
+# Define a function to compute the growth rate based on the cross channel velocity
+
+function growth_rate(model)
+    compute!(v)
+    return norm(interior(v))
+end
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+# FJP 4] Is there a better way to produce a diagnostics file like this?
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (growth_rate = growth_rate,),
+        filepath="growth_rate_shallow_water.nc",
+        schedule=IterationInterval(1),
+        dimensions=(growth_rate=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visiualize the results
+#
+
+using NCDatasets, Plots, IJulia;
```

Comment:
Does the example need IJulia?

Looks like it might be causing the example to fail to build: https://buildkite.com/clima/oceananigans/builds/1544#263b9e4d-3038-43ce-8dab-7389b324299d/40-102

Reply:
Whoops!  Not really but when I didn't include it the animation seemed to freeze, for reasons that I could not figure out.  In order to get it to run I can add IJulia into the docs project and upload that.  Does that sound reasonable to you?

Reply:
Ah the other examples use `@animate` so IJulia might help on your local machine. I always have issues with Plots.jl lol.

I guess I would try removing `using IJulia` from the script and if the movie in the docs doesn't look correct then we should add it back in (and in the `docs/Project.toml`).

Reply:
Movie should work...

Reply:
I have removed the mention of IJulia and things seem to work nicely.  Sorry for including this.